specifyweb/frontend/js_src/lib/divisionfieldcbx.js

```diff
@@ -0,0 +1,51 @@
+"use strict";
+
+var $        = require('jquery');
+var _        = require('underscore');
+var Backbone = require('./backbone.js');
+
+var schema = require('./schema.js');
+
+module.exports =  Backbone.View.extend({
+        __name__: "DivisionPicklistCBX",
+        events: {
+            change: 'set'
+        },
+        initialize: function(info) {
+            this.resource = info.resource;
+            this.resource.on('change:fieldname change:tablename change:type', this.render, this);
+            const divisionQuery = new schema.models.Division.LazyCollection();
+            this.divisionsPromise = new Promise((resolve)=>
+                divisionQuery.fetch().then(()=>resolve(divisionQuery.models))
+            );
+            this.divisions = [];
+        },
+        getPickListFields: function() {
+            return this.divisions;
+        },
+        render: function() {
+            if(typeof this.divisionsPromise === 'object'){
+                this.divisionsPromise.then((divisions)=>{
+                    this.divisions = divisions.map((division)=>({
+                        value: division.get('id'),
+                        title: division.get('name')
+                    }));
+                    this.divisionsPromise = undefined;
```

Comment:
Don't do this! One of the main features of promises is that invoking `then` on an already resolved promise will invoke the continuation immediately so you don't have to keep track of the state manually!

Reply:
Oh my god, why didn't I know about this sooner!
I always though that this code is invalid:
```js
const a = Promise.resolve();
a.then(console.log);
a.then(console.log);  // Error?
```

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#:~:text=A%20promise%20can%20participate%20in%20more%20than%20one%20nesting